modules/chip/ChipInput.tsx

```diff
@@ -1,104 +1,166 @@
+  },
+  input: {},
+  chipList: {
+    width: "100%"
   },
-  input: {
-    border: 0,
-    marginLeft: theme.spaceUnit.xs,
-    ...theme.typography.descriptionMedium,
-    flex: 1,
-    minWidth: 80,
-    outline: "none",
-    minHeight: 40
+  chipContainer: {
+    width: "100%",
+    margin: 0
+  },
+  chip: {
+    justifyContent: "space-between"
   }
 });
 
 function ChipInput({
   styles,
   cx,
   className,
-  value,
-  onChange,
+  onChangeChips,
+  singleChip = false,
   allowDuplicates = false,
   wrappedRef,
   parentStylesheet,
+  value,
+  onChange,
   ...props
 }: IProps & WithStylesProps) {
-  const [_list, setList] = useState(value || []);
-  const [inputValue, setInputValue] = useState("");
+  const [inputValue, setInputValue] = useState(value?.toString() ?? "");
+  const { chips: propsChips } = props;
+  const [chips, setChips] = useState(props.chips ?? []);
+  const [focus, setFocus] = useState(chips.length > 0);
+
+  useEffect(() => {
+    setFocus(focus || chips.length > 0);
+  }, [chips, focus]);
 
-  const ChipListStyle = useRegistryWithStyles<IChipListProps<string>>(
+  const inputStyleSheet: ThemeStyleSheetFactory = useMemo(
+    () => theme => ({
+      prefixIcon: singleChip
+        ? {
+            width: "100%",
+            marginRight: 0
+          }
+        : {}
+    }),
+    [singleChip]
+  );
+  const StyledErrorTextInput = useRegistryWithStyles<IErrorTextInputProps>(
+    "ErrorTextInput",
+    inputStyleSheet
+  );
+  const StyledChipList = useRegistryWithStyles<IChipListProps<string>>(
     "ChipList",
     styleSheet
   );
 
   useEffect(() => {
-    setList(value || []);
+    setInputValue(value?.toString() ?? "");
   }, [value]);
 
+  useEffect(() => {
+    if (propsChips !== undefined) {
+      setChips(propsChips);
+    }
+  }, [propsChips]);
+
+  const handleChange = useCallback(
+    newValue => {
+      if (propsChips === undefined) {
+        setChips(newValue);
+      }
+      return onChangeChips?.(newValue);
+    },
+    [propsChips, onChangeChips]
+  );
+
   const handleInput = useCallback(
     key => {
       if (key.key === "Enter" && inputValue.trim() !== "") {
-        const isIncluded = _list.includes(inputValue.trim());
+        const trimmedValue = inputValue.trim();
+        const isIncluded = chips.includes(trimmedValue);
         if (allowDuplicates || !isIncluded) {
-          const newList = _list.concat([inputValue.trim()]);
-          setList(newList);
-
-          if (onChange) {
-            onChange(newList);
-          }
+          const newList = [...chips, trimmedValue];
+          handleChange(newList);
         }
         setInputValue("");
       }
     },
-    [_list, allowDuplicates, inputValue, onChange]
+    [chips, allowDuplicates, inputValue, handleChange]
   );
 
-  const handleChange = useCallback(
-    newList => {
-      setList(newList);
+  const handleChipClick = useCallback(
+    chip => {
+      handleChange(chips.filter(item => item !== chip));
```

Comment:
Isn't it more efficient or at least easier to read with `handleChange(chips.find(item => item === chip));` ?
Also, it theoretically doesn't matter which chip he clicked on if there are duplicates of that same chip right?

Reply:
The handleChange receives the current list of selected chips.
Good catch, on duplicates I'd say we only want the clicked chip to be removed.